### PR TITLE
Prompt for slug in delete add-on/theme form (fixes #920)

### LINKED ISSUE #920: Change delete add-on form
The delete add-on form currently requires you to enter your password. In a world where passwords are all stored on FxA, that's not going to work, we'll have to come up with something different. We could force FxA authentication or just require something else typed in like the add-on GUID or something.

https://www.dropbox.com/s/97n8gnxodackle1/Screenshot%202015-11-13%2011.22.22.png?dl=0

### PR DESCRIPTION
**Pull request by mstriemer**
delete-add-on mov

### CODE REVIEW

**Comment by wagnerand** [comment]
Developers might not know what *slug* means. Is there a better word for it?

**Comment by mstriemer** [comment]
We seem to call it the URL name or the pretty name. I changed it to say `URL name` and included the slug as the placeholder text.

screenshot 2016-02-02 10 32 49

File: apps/devhub/templates/devhub/addons/listing/delete_form.html [hunk 1]
```diff
@@ -21,10 +21,10 @@
   {% endif %}
   <p>
     <label>
-      {{ _('Enter your password to confirm your decision:') }}
-      {# see bug 630126 for why this is called 'password-timestamp' #}
-      <input type="password" name="password-{{ timestamp }}" data-name="password" autocomplete="off" required>
```
Comment (magopian) [inline apps/devhub/templates/devhub/addons/listing/delete_form.html]: So, it seems the `{{ timestamp }}` hack was added in https://github.com/gkoberger/zamboni/commit/7cd02ca0cfdd333d752b1d105d32329febb599ea. Could we get rid of it now? There's the `timestamp` in the context, and some js code that might now be unused? (Maybe we're using the hack elsewhere too though).

File: apps/devhub/templates/devhub/addons/listing/delete_form.html [hunk 2]
```diff
@@ -21,10 +21,10 @@
...
+      {{ _('Enter the URL name to confirm your decision ({slug}):')|fe(slug=addon.slug) }}
     </label>
+    <input class="full-width" type="text" name="slug" autocomplete="off"
+           placeholder="{{ addon.slug }}" pattern="{{ addon.slug }}" required>
```
Comment (magopian) [inline apps/devhub/templates/devhub/addons/listing/delete_form.html]: Nice use of the `pattern` here ;)

File: apps/devhub/tests/test_views.py [hunk 3]
```diff
@@ -756,26 +756,50 @@ def setUp(self):
...
+        r = self.client.post(self.get_url(), {'slug': 'theme-slug'},
                              follow=True)
+        eq_(pq(r.content)('.notification-box').text(),
+            'URL name was incorrect. Add-on was not deleted.')
+        eq_(self.get_addon().exists(), True)
+
+    def test_post_theme(self):
+        theme = self.make_theme()
+        r = self.client.post(
+            theme.get_dev_url('delete'), {'slug': 'theme-slug'}, follow=True)
```
Comment (magopian) [inline apps/devhub/tests/test_views.py]: Why not use `self.get_url()` here? (same below)

File: apps/devhub/tests/test_views.py [hunk 4]
```diff
@@ -3341,14 +3365,14 @@ def setUp(self):
         self.client.login(username='admin@mozilla.com', password='password')
 
     def test_bad_password(self):
-        r = self.client.post(self.url, dict(password='turd'))
+        r = self.client.post(self.url, dict(slug='nope'))
```
Comment (magopian) [inline apps/devhub/tests/test_views.py]: thanks for getting rid of the profanity :)

**Comment by magopian** [comment]
Awesome work!

r+wc

File: apps/devhub/templates/devhub/addons/listing/delete_form.html (hunk 1 above)
Comment (mstriemer) [inline apps/devhub/templates/devhub/addons/listing/delete_form.html]: I initially went to remove it but noticed other files using the JS. The delete account view has a similar password prompt so I'm betting it's that. I'll check again once I remove the other password prompt.

I'll pull `timestamp` out of the context though, good call.

File: apps/devhub/tests/test_views.py (hunk 3 above)
Comment (mstriemer) [inline apps/devhub/tests/test_views.py]: `self.get_url()` is for the add-on from the fixture. It was previously just changing the type and only tested the negative case but you can't actually delete an add-on that is pretending to be a theme. It raises an error about not having a `Persona`.

I updated the method to take the add-on and then realised that I could just call a method on the add-on itself and it didn't really make sense anymore.

File: apps/devhub/tests/test_views.py (hunk 4 above)
Comment (mstriemer) [inline apps/devhub/tests/test_views.py]: Heh, I was a little surprised when I saw this. Not that it's particularly bad but not what I was expecting to see.

Comment (magopian) [inline apps/devhub/tests/test_views.py]: ah, sure, fair enough ;)

**Pull request merged**